mirest/ah-haven-space-sprinters
# #164694242 Like or unlike a comment

#### What does this PR do?
* Add like/unlike a specific comment functionality

#### Description of Task to be completed?

- [x] add comment-like model

- [x] add comment-like views and relevant methods

- [x] write unittests

#### How should this be manually tested?
You can test locally by running command `python manage.py runserver`
- Log in using verified credentials
- Go to article of your choice, find comments attached to that article using this endpoint`GET /api/articles/:slug/comments`
- To like a specific comment you will have to use this  `PUT /api/articles/:slug/comments/<int:comment_id>/like/` endpoint.
-To unlike a comment that you liked, you will have to use this `DELETE /api/articles/:slug/comments/<int:comment_id>/like/` endpoint. 
- Note that we do not have to pass any request data for the above endpoints. In order for you to dislike a comment, you should have liked it first

#### Any background context you want to provide?
Besides Authors getting feedback on articles from readers, we would like to receive feedback on the comments left against an article.

#### What are the relevant pivotal tracker stories?
[#164694242](https://www.pivotaltracker.com/story/show/164694242)

#### Relevant screenshots
1. like a comment
![like a commeny initially](https://user-images.githubusercontent.com/9926422/56710111-36144c80-672d-11e9-9a4e-2d8a9180d1e1.png)

2. Like a comment, you have already liked
![like a comment](https://user-images.githubusercontent.com/9926422/56710130-4d533a00-672d-11e9-857b-2d230f53497f.png)

3. unlike a comment
![unlike comment](https://user-images.githubusercontent.com/9926422/56710149-665beb00-672d-11e9-9c9f-0b23f5d2c8d3.png)
